### PR TITLE
Implement --version flag

### PR DESCRIPTION
**Pull request by mrswats**
As mentioned in #142 implements the version flag for this project.

### CODE REVIEW

File: src/django_upgrade/main.py [hunk 1]
```diff
@@ -22,6 +23,11 @@ def main(argv: Sequence[str] | None = None) -> int:
     parser = argparse.ArgumentParser()
     parser.add_argument("filenames", nargs="*")
     parser.add_argument("--exit-zero-even-if-changed", action="store_true")
+    parser.add_argument(
```
Comment (adamchainz) [inline src/django_upgrade/main.py:26]: I like things in alphabetical order when there's no other ordering enforced :)

File: tests/test_main.py [hunk 2]
```diff
@@ -24,6 +25,17 @@ def test_main_help():
...
 
 
+def test_main_version(capsys):
+    with pytest.raises(SystemExit) as excinfo:
+        main(["--version"])
+
+    out, err = capsys.readouterr()
+
+    assert excinfo.value.code == 0
+    assert re.match(r"\d\.\d\.\d", out)
```
Comment (adamchainz) [inline tests/test_main.py:35]: `re.fullmatch` is a better test :)

File: src/django_upgrade/main.py [hunk 3]
```diff
@@ -22,6 +23,11 @@ def main(argv: Sequence[str] | None = None) -> int:
...
+        "--version",
+        action="version",
+        version=metadata.version("django-upgrade"),
```
Comment (adamchainz) [inline src/django_upgrade/main.py:29]: I guess it's best to use `%(prog)s` in the version string, as the argparse docs demonstrate for `action='version'`.

**Review by adamchainz — COMMENTED** [review]
I'm gonna push a few tweaks based on the below comments

**Comment by mrswats** [comment]
I was implementing the changes already but okay :)

**Pull request merged**